src/Extensions/DibiConnectionFetchDynamicReturnTypeExtension.php

```diff
@@ -118,6 +118,10 @@ private function reduceResultType(MethodReflection $methodReflection, Type $resu
         } elseif ('fetchPairs' === $methodName && $resultType instanceof ConstantArrayType && 2 === \count($resultType->getValueTypes())) {
             return new ArrayType($resultType->getValueTypes()[0], $resultType->getValueTypes()[1]);
         } elseif ('fetchSingle' === $methodName && $resultType instanceof ConstantArrayType && 1 === \count($resultType->getValueTypes())) {
+            if ($resultType->getValueTypes()[0]->accepts(new NullType(), true)->yes()) {
+                return $resultType->getValueTypes()[0];
+            }
+
             return new UnionType([new NullType(), $resultType->getValueTypes()[0]]);
```

Comment:
I guess the newly added lines are unnecessary and this should work with `TypeCombinator::addNull()` instead

Reply:
yes, I wasnt aware of such method. That solves it too. Will push changes shortly

Reply:
hopefully fixed